File: src/sdk/java/src/eu/venusc/cdmi/BlobOperations.java
package eu.venusc.cdmi;
import java.util.List;

public interface BlobOperations  {
 
	// Create
	public void create(String localFile, String remoteFNM) throws Exception;
	
	public void delete(String remoteFNM) throws Exception;

	public void update(String localFile, String remoteFNM) throws Exception;
	
	public int readFile(String remoteFNM) throws Exception;

	public List<Blob> list(String remoteContainer);
	
}

File: src/sdk/python/vcdm/BlobOperations.py
import urllib2

class BlobOperations():
    
    credentials = None
    endpoint = None
    
    def __init__(self, credentials, endpoint):
        self.credentials = credentials
        self.endpoint = endpoint
            
    def create(self, remotefile, localfile,  mimetype=None, metadata=None):
        """Create a new blob from a local file"""
        pass        
    
    def update(self, remoteblob, localfile = None, mimetype=None, metadata=None):
        """Update a remote blob with new data."""
        pass
    
    def read(self, remoteblob):
        """Read contents of a blob"""
        pass
    
    def delete(self, remoteblob):
        """Delete specified blob"""
        pass